libs/remote_storage/src/s3_bucket.rs

```diff
@@ -144,7 +144,7 @@ impl S3Bucket {
         assert_eq!(std::path::MAIN_SEPARATOR, REMOTE_STORAGE_PREFIX_SEPARATOR);
         let path_string = path
             .get_path()
-            .to_string_lossy()
+            .to_string()
             .trim_end_matches(REMOTE_STORAGE_PREFIX_SEPARATOR)
             .to_string();
```

Comment:
Now noticed that this looks odd.

First `to_string` should probably be `Utf8PathBuf::as_str`.

Second `to_string` is not needed at all.

This is semi-unrelated, but I it was required before to use the `+` operator because before the `to_string_lossy` returned a `Cow<'_, str>`..

Reply:
Yes, it is right that first `to_string` is, in fact, `as_str`.
But, we need second `to_string` since return type is `String`.
Or we can add this `to_string` in `match::None` statement.